File: Ejercicios Propuestos 2/Ejercicio 1/Ejercicio 1.jl
# Bucle de validación hasta que se ingrese un número válido
while true
    println("Ingrese un número entero positivo (mayor o igual a 1):")
    limite = parse(Int, readline())
    
    # Si el número es incorrecto, se muestra error
    if limite < 1
        println("¡Error! El número ingresado no es válido.")
        println("Por favor, intente de nuevo con un número positivo.")
        println("--------------------------------------------------")
    else
        global limite
        break # Hasta Que limite >= 1
    end
end

factorial = 1
println("--------------------------------------------------")
println("Los factoriales hasta el ", limite, " son:")

# Para calcular e imprimir 
for i in 1:limite
    global factorial
    factorial = factorial * i
    println("El factorial de ", i, " es: ", factorial)
end

File: Ejercicios Propuestos 2/Ejercicio 1/Ejercicio 1.java
import java.util.Scanner;

public class EJERCICIODeGRUPO2 {
    public static void main(String[] args) {

        Scanner sc = new Scanner(System.in);

        int limite;
        int i;
        long factorial = 1;

        // Validación del número
        do {
            System.out.println("Ingrese un número entero positivo (>= 1):");
            limite = sc.nextInt();

            if (limite < 1) {
                System.out.println("Error: número no válido.");
                System.out.println("Intente nuevamente.");
                System.out.println("-------------------------");
            }

        } while (limite < 1);

        System.out.println("-------------------------");
        System.out.println("Los factoriales hasta " + limite + " son:");

        // Cálculo de factoriales sucesivos
        for (i = 1; i <= limite; i++) {
            factorial = factorial * i;
            System.out.println("Factorial de " + i + " = " + factorial);
        }

        sc.close();
    }
}
